app/models/step.rb

```diff
@@ -33,6 +34,10 @@ def pre_order_tree_traversal
     [self] + self.child_approvals.flat_map(&:pre_order_tree_traversal)
   end
 
+  def completer
```

Comment:
We already have a `approved_at` timestamp column. AFAIK the plan is to rename it to `completed_at` in some future naming cleanup as part of the whole `Approval` => `Step` conversion.

Our problem is not detecting when/if a step is completed. Our problem is that 2/3 of NCR approvals are assigned to non-human mailbox Users, and then actually completed via real-human delegate Users, and we want to preserve the original non-human link as well as the actual real-human link so that authorization is preserved.

I.e., for 100% of Tier 1 and Tier 2 approvals, the 2 foreign keys will go to 2 different User records.